src/App.tsx

```diff
@@ -1,61 +1,110 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+const findUserById = (todoUserId: Todo['userId'], users:User[]) => {
+  return users.find(user => user.id === todoUserId) || null;
+};
+
+const todoWithUsers = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(todo.userId, usersFromServer),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [isValidTitle, setIsValidTitle] = useState(true);
+  const [selectedOption, setSelectedOption] = useState(0);
+  const [isValidSelectedOption, setIsValidSelectedOption] = useState(true);
+  const [todos, setTodos] = useState(todoWithUsers);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsValidTitle(true);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedOption(+event.target.value);
+    setIsValidSelectedOption(true);
+  };
+
+  const addNewTodo = (() => setTodos(prevTodos => [...prevTodos, {
+    id: Math.max(...prevTodos.map(todo => todo.id)) + 1,
+    title,
+    completed: false,
+    userId: selectedOption,
+    user: findUserById(selectedOption, usersFromServer),
+  }]));
+
+  const handleFormSubmit = (
+    (event:React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (title.length && selectedOption !== 0) {
+        addNewTodo();
+        setTitle('');
+        setSelectedOption(0);
+      } else {
+        if (selectedOption === 0) {
+          setIsValidSelectedOption(false);
+        }
+
+        if (title.length === 0) {
+          setIsValidTitle(false);
+        }
+      }
```

Comment:
try to come up with a better solution for this case🤔

I think we can remove nested if statement.